processing/src/main/java/org/apache/druid/query/expression/IPv4AddressMatchExprMacro.java

```diff
@@ -71,71 +73,75 @@ public Expr apply(final List<Expr> args)
       throw new IAE(ExprUtils.createErrMsg(name(), "must have 2 arguments"));
     }
 
-    IPAddressString subnetInfo = getSubnetInfo(args);
-    Expr arg = args.get(0);
+    try {
+      final Expr arg = args.get(0);
+      final IPAddressString blockString = getSubnetInfo(args);
+      final IPAddress block = blockString.toAddress().toPrefixBlock();
```

Comment:
maybe a comment could be useful to tell about the different ways to match a CIDR prefix being used (one with prefixBlock address and one directly with blockString#prefixContains)